be1-go/internal/hub/hub.go

```diff
@@ -242,23 +246,35 @@ func (h *Hub) SendGreetServer(socket socket.Socket) error {
 
 func (h *Hub) runMessageReceiver() {
 	defer h.wg.Done()
-	defer logger.Logger.Info().Msg("stopping hub")
+	defer h.log.Info().Msg("stopping hub")
 
-	logger.Logger.Info().Msg("starting hub")
+	h.log.Info().Msg("starting hub")
 
 	for {
-		logger.Logger.Debug().Msg("waiting for a new message")
+		h.log.Debug().Msg("waiting for a new message")
 		select {
 		case incomingMessage := <-h.messageChan:
-			logger.Logger.Debug().Msg("start handling a message")
+			h.log.Debug().Msg("start handling a message")
 			err := h.jsonRpcHandler.Handle(incomingMessage.Socket, incomingMessage.Message)
 			if err != nil {
-				logger.Logger.Error().Err(err)
+				popError := &poperrors.PopError{}
+				if !errors.As(err, &popError) {
+					popError = poperrors.NewPopError(-6, err.Error())
+				}
```

Comment:
Maybe changing the `-6` to `poperrors.InternalServerErrorCode` could be easier to remember what code it is for.

Using err.Error() directly as the format string could potentially result in a format string attack ?

Reply:
done